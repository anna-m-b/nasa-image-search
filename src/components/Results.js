import React from 'react';
import ResultImage from './ResultImage';
import '../styles/Results.css';

const Results = ({ results }) => {
  console.log('results', results)
  const display = results.map((item) => {
    const thumb = item.links.find((link) => /thumb/.test(link));
    const original = item.links.find((link) => /orig/.test(link));
    return (
      <ResultImage
        url={thumb}
        description={item.title}
        key={item.nasa_id}
        linkToOriginal={original}
        data-testid="result-image"
      />
    );
  });

  return <div className="results" data-testid="results">{display}</div>;
};

export default Results;
