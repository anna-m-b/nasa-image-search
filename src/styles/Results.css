.results {
  width: 90vw;
  margin: 50px auto;
  columns: 4 200px;
}

