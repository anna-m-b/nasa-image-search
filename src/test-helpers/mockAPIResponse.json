{
  "data": {
    "collection": {
      "href": "https://images-api.nasa.gov/search?media_type=image&q=moon",
      "items": [
        {
            "href": "https://images-assets.nasa.gov/image/PIA12235/collection.json",
            "links": [
                {
                    "href": "https://images-assets.nasa.gov/image/PIA12235/PIA12235~thumb.jpg",
                    "render": "image",
                    "rel": "preview"
                }
            ],
            "data": [
                {
                    "center": "JPL",
                    "media_type": "image",
                    "description": "Nearside of the Moon",
                    "nasa_id": "PIA12235",
                    "secondary_creator": "ISRO/NASA/JPL-Caltech/Brown Univ.",
                    "keywords": [
                        "Moon",
                        "Chandrayaan-1"
                    ],
                    "description_508": "Nearside of the Moon",
                    "title": "Nearside of the Moon",
                    "date_created": "2009-09-24T18:00:22Z"
                }
            ]
        },
        {
            "href": "https://images-assets.nasa.gov/image/PIA13517/collection.json",
            "links": [
                {
                    "href": "https://images-assets.nasa.gov/image/PIA13517/PIA13517~thumb.jpg",
                    "render": "image",
                    "rel": "preview"
                }
            ],
            "data": [
                {
                    "center": "JPL",
                    "media_type": "image",
                    "description": "Color of the Moon",
                    "nasa_id": "PIA13517",
                    "secondary_creator": "NASA/GSFC/Arizona State University",
                    "keywords": [
                        "Moon",
                        "Lunar Reconnaissance Orbiter LRO"
                    ],
                    "description_508": "Color of the Moon",
                    "title": "Color of the Moon",
                    "date_created": "2010-09-10T22:24:40Z"
                }
            ]
        },
        {
            "href": "https://images-assets.nasa.gov/image/PIA12228/collection.json",
            "links": [
                {
                    "href": "https://images-assets.nasa.gov/image/PIA12228/PIA12228~thumb.jpg",
                    "render": "image",
                    "rel": "preview"
                }
            ],
            "data": [
                {
                    "center": "JPL",
                    "media_type": "image",
                    "description": "Cassini Look at Water on the Moon",
                    "nasa_id": "PIA12228",
                    "secondary_creator": "NASA/JPL-Caltech/USGS",
                    "keywords": [
                        "Moon",
                        "Cassini-Huygens"
                    ],
                    "description_508": "Cassini Look at Water on the Moon",
                    "title": "Cassini Look at Water on the Moon",
                    "date_created": "2009-09-24T18:00:15Z"
                }
            ]
        },
        {
            "href": "https://images-assets.nasa.gov/image/PIA12229/collection.json",
            "links": [
                {
                    "href": "https://images-assets.nasa.gov/image/PIA12229/PIA12229~thumb.jpg",
                    "render": "image",
                    "rel": "preview"
                }
            ],
            "data": [
                {
                    "center": "JPL",
                    "media_type": "image",
                    "description": "Mineral Mapping the Moon",
                    "nasa_id": "PIA12229",
                    "secondary_creator": "ISRO/NASA/JPL-Caltech/Brown Univ.",
                    "keywords": [
                        "Moon",
                        "Chandrayaan-1"
                    ],
                    "description_508": "Mineral Mapping the Moon",
                    "title": "Mineral Mapping the Moon",
                    "date_created": "2009-09-24T18:00:16Z"
                }
            ]
        },
        {
            "href": "https://images-assets.nasa.gov/image/PIA12233/collection.json",
            "links": [
                {
                    "href": "https://images-assets.nasa.gov/image/PIA12233/PIA12233~thumb.jpg",
                    "render": "image",
                    "rel": "preview"
                }
            ],
            "data": [
                {
                    "center": "JPL",
                    "media_type": "image",
                    "description": "Mapping the Moon, Point by Point",
                    "nasa_id": "PIA12233",
                    "secondary_creator": "NASA/JPL-Caltech",
                    "keywords": [
                        "Moon",
                        "Chandrayaan-1"
                    ],
                    "description_508": "Mapping the Moon, Point by Point",
                    "title": "Mapping the Moon, Point by Point",
                    "date_created": "2009-09-24T18:00:20Z"
                }
            ]
        }
      ],
      "version": "1.0",
      "metadata": {
        "total_hits": 9941
      },
      "links": [
        {
          "href": "https://images-api.nasa.gov/search?media_type=image&q=moon&page=2",
          "rel": "next",
          "prompt": "Next"
        }
      ]
    }
  }
}
