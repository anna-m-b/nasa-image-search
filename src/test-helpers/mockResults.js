export const mockResults = [
  {
    media_type: 'image',
    nasa_id: 'PIA12235',
    title: 'Nearside of the Moon',
    thumb: 'https://images-assets.nasa.gov/image/PIA12235/PIA12235~thumb.jpg',
  },
  {
    media_type: 'image',
    nasa_id: 'PIA13517',
    title: 'Color of the Moon',
    thumb: 'https://images-assets.nasa.gov/image/PIA13517/PIA13517~thumb.jpg',
  },
  {
    media_type: 'image',
    nasa_id: 'PIA12228',
    title: 'Cassini Look at Water on the Moon',
    thumb: 'https://images-assets.nasa.gov/image/PIA12228/PIA12228~thumb.jpg',
  },
  {
    media_type: 'image',
    nasa_id: 'PIA12229',
    title: 'Mineral Mapping the Moon',
    thumb: 'https://images-assets.nasa.gov/image/PIA12229/PIA12229~thumb.jpg',
  },
  {
    media_type: 'image',
    nasa_id: 'PIA12233',
    title: 'Mapping the Moon, Point by Point',
    thumb: 'https://images-assets.nasa.gov/image/PIA12233/PIA12233~thumb.jpg',
  },
];
