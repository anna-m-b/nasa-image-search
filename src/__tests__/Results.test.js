import React from 'react';
import { render } from '@testing-library/react';
import { mockResults } from '../test-helpers/mockResults';
import Results from '../components/Results';

describe('Results', () => {
  test('renders correctly', () => {
    const { asFragment } = render(<Results results={mockResults} />)
    expect(asFragment()).toMatchSnapshot();
  })
  test('search results are displayed as images', () => {
    const { getAllByTestId } = render(<Results results={mockResults} />);
    expect(getAllByTestId('result-image')).toHaveLength(5);
  });
});
