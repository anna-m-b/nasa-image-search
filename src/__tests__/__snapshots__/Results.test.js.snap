// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results renders correctly 1`] = `
<DocumentFragment>
  <div
    class="results"
    data-testid="results"
  >
    <div
      class="result-image"
    >
      <a
        href="http://images-assets.nasa.gov/image/PIA18906/PIA18906~orig.jpg"
      >
        <img
          alt="Sun Shines in High-Energy X-rays"
          class="result-image_img"
          data-testid="result-image"
          src="http://images-assets.nasa.gov/image/PIA18906/PIA18906~thumb.jpg"
        />
      </a>
    </div>
    <div
      class="result-image"
    >
      <a
        href="http://images-assets.nasa.gov/image/PIA15179/PIA15179~orig.jpg"
      >
        <img
          alt="The Sun Magnetic Field"
          class="result-image_img"
          data-testid="result-image"
          src="http://images-assets.nasa.gov/image/PIA15179/PIA15179~thumb.jpg"
        />
      </a>
    </div>
    <div
      class="result-image"
    >
      <a
        href="http://images-assets.nasa.gov/image/PIA14729/PIA14729~orig.jpg"
      >
        <img
          alt="Mercury Transit Across the Sun"
          class="result-image_img"
          data-testid="result-image"
          src="http://images-assets.nasa.gov/image/PIA14729/PIA14729~thumb.jpg"
        />
      </a>
    </div>
    <div
      class="result-image"
    >
      <a
        href="http://images-assets.nasa.gov/image/PIA17669/PIA17669~orig.jpg"
      >
        <img
          alt="Pulses from the Sun"
          class="result-image_img"
          data-testid="result-image"
          src="http://images-assets.nasa.gov/image/PIA17669/PIA17669~thumb.jpg"
        />
      </a>
    </div>
    <div
      class="result-image"
    >
      <a
        href="http://images-assets.nasa.gov/image/PIA09771/PIA09771~orig.jpg"
      >
        <img
          alt="Filtering the Sun"
          class="result-image_img"
          data-testid="result-image"
          src="http://images-assets.nasa.gov/image/PIA09771/PIA09771~thumb.jpg"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;
